src/util/animation/ColorAnimation.ts

```diff
@@ -1,26 +1,36 @@
 import { Color } from '../../color/color.class';
 import { TRGBAColorSource } from '../../color/color.class';
+import { halfPI } from '../../constants';
 import { capValue } from '../misc/capValue';
 import { AnimationBase } from './AnimationBase';
-import { ColorAnimationOptions, TOnAnimationChangeCallback } from './types';
+import type {
+  ColorAnimationOptions,
+  TEasingFunction,
+  TOnAnimationChangeCallback,
+} from './types';
+
+const defaultColorEasing: TEasingFunction = (
+  timeElapsed,
+  startValue,
+  byValue,
+  duration
+) => {
+  const durationProgress = 1 - Math.cos((timeElapsed / duration) * halfPI);
+  return startValue + byValue * durationProgress;
+};
 
 const wrapColorCallback = <R>(
   callback?: TOnAnimationChangeCallback<string, R>
 ) =>
   callback &&
-  ((rgba: TRGBAColorSource, valueRatio: number, durationRatio: number) =>
-    callback(new Color(rgba).toRgba(), valueRatio, durationRatio));
+  ((rgba: TRGBAColorSource, valueProgress: number, durationProgress: number) =>
+    callback(new Color(rgba).toRgba(), valueProgress, durationProgress));
```

Comment:
I also think that the the color animation callbacks should pass a `Color` instance rather that a color string
```suggestion
    callback(new Color(rgba), valueProgress, durationProgress));
```

Reply:
> > The only question remaining is should we merge between animate and animateColor?
> 
> If we change color animation options to accept only a color instance (which makes sense and is not that breaking) we could merge it into `animate` w/o a problem by checking `instance of Color` and extend the `isArrayAnimation` functionality

https://github.com/fabricjs/fabric.js/compare/small-animation-preferences...animate-color